### DISCUSSION TITLE
snippets dynamic level adjustment

### DISCUSSION THREAD

**Discussion by csantanapr** [squidfunk/mkdocs-material#3581] [Q&A]
We are currently using [snippet](https://squidfunk.github.io/mkdocs-material/setup/extensions/python-markdown-extensions/#snippets)

Do we wonder if there is a way to adjust the headings (ie H2, H3, H4) of the text being inserted?

We want to add the text, but in some pages, it will be under an H2 `##` and some cases under H3 `###` 

The text itself have have its headings as H2 `##` we wonder if there is a plugin that will adjust the headings by making them match as a child to the heading of the parent page.

Someone found this extension https://github.com/cmacmackin/markdown-include#configuration
We wonder if someone have use it with material-mkdocs with success

**Comment by squidfunk** [comment]
__Related__

- https://github.com/facelessuser/pymdown-extensions/issues/1052

**Reply by facelessuser** [reply]
Yeah, I haven't spent any time on dynamic header adjustments yet in Snippets. I have some ideas, but it has currently been a lower so far.

**Reply by abrennan89** [reply]
Any update on this one?

**Reply by facelessuser** [reply]
No update. I support a number of repositories, and this issue has been a lower priority one. With that said, Snippets has been the focus of my work in Pymdown Extensions recently. It is likely that I will be able to further evaluate the request soon and make a decision.
